src/lib/rnp.cpp

```diff
@@ -6173,6 +6173,64 @@ try {
 }
 FFI_GUARD
 
+rnp_result_t
+rnp_key_get_default_key(rnp_key_handle_t  primary_key,
+                        const char *      usage,
+                        uint32_t          flags,
+                        rnp_key_handle_t *default_key)
+try {
+    if (!primary_key || !usage || !default_key) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    uint8_t keyflag = 0;
+    bool    no_primary = false;
+    if (!str_to_key_flag(usage, &keyflag)) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    if (flags & RNP_KEY_SUBKEYS_ONLY) {
+        no_primary = true;
+        flags &= ~RNP_KEY_SUBKEYS_ONLY;
+    }
+    if (flags) {
```

Comment:
I wondering whether should we add RNP_KEY_SECRET_ONLY, or (better?) search for secret key automatically with `sign`/`certify` usage?

Reply:
> or (better?) search for secret key automatically with sign/certify usage?

I think second option is better

Reply:
@ni4 what condition should be used in this case, both public & secret must be available or only secret is enough?

Reply:
Only secret is enough, public key may be easily re-generated from secret (and actually must correspond to it, except secret fields availability).

Reply:
Looks like find_suitable_key() should be modified then to include this option, there can be situation where primary key & some suitable subkeys are presented only by public part, but some other suitable subkey can have private part and not be selected as default according to current implementation.

Reply:
@antonsviridenko Agree, some parameter like `is_secret` could be added to `find_suitable_key()`.

Reply:
ok, I've implemented that parameter and now this new function returns expired subkey from test keyring # 1:
```
ssb   1024/DSA 1d7e8a5393c997a8 2017-07-20 [S] [EXPIRED 2017-11-20]
      c5b15209940a7816a7af3fb51d7e8a5393c997a8
```
Probably expiration is not checked somewhere for secret keys.

Reply:
@antonsviridenko Yeah, initially secret keys were considered as valid as owner may always re-generate self-signatures. Right now I think that this should be changed, as may give some difficulties and problems. It's in a my pipeline.
However, adding FFI function should not change old behaviour as may bring some problems to already existing users. So probably for now let's leave the case with non-existing secret primary/subkeys (actually, right now we do not support public primary + secret subkey, we support only secret primary with special s2k, where secret key data is cut out).

Reply:
@ni4 searching for secret keys breaks the test (which passed before), see my latest commit. So either we should revert this idea about secret keys or check for secret part presence on the key returned by original find_suitable_key() implemenation.

Reply:
i.e. we still want to find non-expired keys, but their secret part should be present for specific usages (other than encrypt in this case, all others require secret key)

Reply:
@antonsviridenko Within current implementation secret key is required twice - on the `find_suitable_key()` level and in the `rnp_key_get_default_key()`. You may rollback changes to the `find_suitable_key()`, so it will behave as previously, and require secret only on the `rnp_key_get_default_key()` level. Basically, we should first find key/subkey which is good enough by flags/validity, and then find it's secret part or return an error.